### PR TITLE
add npair loss op

### PR DESCRIPTION
**Pull request by ceci3**
http://www.nec-labs.com/uploads/images/Department-Images/MediaAnalytics/papers/nips16_npairmetriclearning.pdf

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_npair_loss_op.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
```
Comment (wanghaoshuang) [inline python/paddle/fluid/tests/unittests/test_npair_loss_op.py]: 2018->2019

**Review by wanghaoshuang — COMMENTED** [review]
还要修改下API.spec文件

File: python/paddle/fluid/layers/nn.py [hunk 2]
```diff
@@ -10340,3 +10341,51 @@ def tree_conv(nodes_vector,
...
+    l2loss = reduce_mean(reduce_sum(square(anchor), 1)) \
+             + reduce_mean(reduce_sum(square(positive), 1))
+
+    similarity_matrix = matmul(
+        anchor, positive, transpose_x=False, transpose_y=True)
+    softmax_value = softmax(similarity_matrix)
+    cross_entropy = -1 * reduce_sum(labels * log(softmax_value), 0)
+    celoss = reduce_mean(cross_entropy)
+
+    return l2loss * 0.25 * l2_reg + celoss
```
Comment (wanghaoshuang) [inline python/paddle/fluid/layers/nn.py]: 0.25这种数字最好声明为常量
